ONSdigital/dp-frontend-dataset-controller
# Implement health endpoint with checkers

### What

Implement new health check library, includes checkers for frontend renderer, zebedee, dataset api and filter api.

Also includes:

* Instantiate config.Config once and copy object for use across handlers; some handlers would create a new config based on the environments exported variables every time a request is made; not only is this bad if it errors but the error handling would initiate a shutdown of the application (and not gracefully either)

* Update dependency manager to go modules and use latest versions of libraries
* Switch out `go-ns/log` package for `log.go/log`
* Updates to documentation, ci build, license etc.

### How to review

* Check changes make sense
* Unit tests pass
* Run application without any dependencies and run `curl localhost:20200/health` that should return status warning with health check object (after critical timeout this status will change to critical)
* Run application with dependencies on and run `curl localhost:20200/health` this should return status OK with all checks returning OK - play around with switching services on and off to see how the health checks get reflected and the impact on the overall status of this app

Upstream dependencies:
* zebedee
* frontend renderer
* dataset api
* filter api

* Lastly check the application works as expected - other endpoints beside `/health` are reacting accordingly (may need to be tested by a frontend developer?)


### Who can review

Anyone
